sftp/sftp.go

```diff
@@ -508,9 +508,9 @@ func (o *Object) Hash(r fs.HashType) (string, error) {
 	err = fs.ErrHashUnsupported
 	var outputBytes []byte
 	if r == fs.HashMD5 {
-		outputBytes, err = session.Output("md5sum " + o.path() + " | awk '{ print $1 }'")
+		outputBytes, err = session.Output("md5sum \"" + o.path() + "\" | awk '{ print $1 }'")
```

Comment:
I gave `quotemeta` a try, but it doesn't very portable. It escapes single dots (`.`) as `\.`, which seems to be recognized by some shells, but not all. When I ran this, it made `md5sum`/`sha1sum` not find the file.

Can file names even contain any characters which break simple double quote escaping? If so, [printf](https://stackoverflow.com/a/5608358/1502352) could potentially to the trick.